backpack/extensions/secondorder/diag_ggn/__init__.py

```diff
@@ -132,3 +132,59 @@ def __init__(self, mc_samples=1):
 
     def get_num_mc_samples(self):
         return self._mc_samples
+
+
+class BatchDiagGGN(BackpropExtension):
+    VALID_LOSS_HESSIAN_STRATEGIES = [
+        LossHessianStrategy.EXACT,
+        LossHessianStrategy.SAMPLING,
+    ]
+
+    def __init__(self, loss_hessian_strategy, savefield):
+        self.loss_hessian_strategy = loss_hessian_strategy
+        super().__init__(
+            savefield=savefield,
+            module_exts={
+                MSELoss: losses.DiagGGNMSELoss(),
+                CrossEntropyLoss: losses.DiagGGNCrossEntropyLoss(),
+                Linear: linear.BatchDiagGGNLinear(),
+                MaxPool1d: pooling.DiagGGNMaxPool1d(),
+                MaxPool2d: pooling.DiagGGNMaxPool2d(),
+                AvgPool1d: pooling.DiagGGNAvgPool1d(),
+                MaxPool3d: pooling.DiagGGNMaxPool3d(),
+                AvgPool2d: pooling.DiagGGNAvgPool2d(),
+                AvgPool3d: pooling.DiagGGNAvgPool3d(),
+                ZeroPad2d: padding.DiagGGNZeroPad2d(),
+                Conv1d: conv1d.BatchDiagGGNConv1d(),
+                Conv2d: conv2d.BatchDiagGGNConv2d(),
+                Conv3d: conv3d.BatchDiagGGNConv3d(),
+                ConvTranspose1d: convtranspose1d.BatchDiagGGNConvTranspose1d(),
+                ConvTranspose2d: convtranspose2d.BatchDiagGGNConvTranspose2d(),
+                ConvTranspose3d: convtranspose3d.BatchDiagGGNConvTranspose3d(),
+                Dropout: dropout.DiagGGNDropout(),
+                Flatten: flatten.DiagGGNFlatten(),
+                ReLU: activations.DiagGGNReLU(),
+                Sigmoid: activations.DiagGGNSigmoid(),
+                Tanh: activations.DiagGGNTanh(),
+                LeakyReLU: activations.DiagGGNLeakyReLU(),
+                LogSigmoid: activations.DiagGGNLogSigmoid(),
+                ELU: activations.DiagGGNELU(),
+                SELU: activations.DiagGGNSELU(),
```

Comment:
Can we recycle those classes, or will they fetch and backpropagate the 'wrong' quantity?